ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -463,114 +453,32 @@ private DdlCommand createDdlCommand(
           statementText,
           serviceContext.getSchemaRegistryClient());
     }
-  }
-
-  private static final class EngineParser {
```

Comment:
The engine parsing steps now either don't require the metastore, or don't mutate the metastore, much simplifying the code and remove the duplicate logic.  So this inner class is no longer needed.